### ISSUE TITLE
Retraining vocoder (hifigan) and dvae in XTTS

### ISSUE THREAD

**Issue by cnlinxi** [coqui-ai/TTS#3323]
**🚀 Feature Description**

Thank you for your contributing!

What is the way to retrain HiFiGAN and DVAE in XTTS v2.0? If I want to use a new training set, but these audio only is in 16kHz sampling rate, do these two parts need to be retrained?

**Comment by dy2009** [comment]
I also need re-train HiFiGAN and DVAE in XTTS v2.0, If you have code, Please tell me.

**Comment by dy2009** [comment]
> **🚀 Feature Description**
> 
> Thank you for your contributing!
> 
> What is the way to retrain HiFiGAN and DVAE in XTTS v2.0? If I want to use a new training set, but these audio only is in 16kHz sampling rate, do these two parts need to be retrained?

I have complished the hifi-gan and dvae train code.

**Comment by OnceJune** [comment]
> > **🚀 Feature Description**
> > Thank you for your contributing!
> > What is the way to retrain HiFiGAN and DVAE in XTTS v2.0? If I want to use a new training set, but these audio only is in 16kHz sampling rate, do these two parts need to be retrained?
> 
> I have complished the hifi-gan and dvae train code.

Cool, any plan to open source?

**Comment by ericwudayi** [comment]
Hi @dy2009 , This is a very cool and useful feature! Hoping can open source it also!

I found the input feature of hifidecoder is gpt latent space, but not code indices itself. I am wondering how to re-train the hifigan model.

**Comment by zhangyue678** [comment]
I want to use a new training set, and these audio only is in 16kHz sampling rate, do these two parts need to be retrained?

**Comment by Liujingxiu23** [comment]
> Hi @dy2009 , This is a very cool and useful feature! Hoping can open source it also!
> 
> I found the input feature of hifidecoder is gpt latent space, but not code indices itself. I am wondering how to re-train the hifigan model.

Have you know hwo to train of finetune the hifidecoder?

**Comment by Yaodada12** [comment]
大佬们有训练出来吗，我准备微调speaker_encoder，有啥思路吗

**Comment by Liujingxiu23** [comment]
Does any one retrain the model successfullly? How to do this?

**Comment by Liujingxiu23** [comment]
when you finetune the GPT model of XTTS v2, how about the loss? loss_ce is about ?

**Comment by tuanh123789** [comment]
I think I have completed the hifidecoder training part. The results were quite good when testing with the Ljspeech dataset. I will push it to github soon, stay tuned

**Comment by tuanh123789** [comment]
The code available here: https://github.com/tuanh123789/Train_Hifigan_XTTS
